### PR TITLE
Fix(medusa): Missing location id on fulfillments

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- include location id when creating a fulfillment
- Allow location updates to reservations without passing along quantity

**Why**
- location_id on fulfillment was null after creation

Fixes CORE-1242, CORE-1243

### CODE REVIEW

**Comment by srindom** [comment]
@pKorsholm - any Linear tickets to link?

File: integration-tests/plugins/__tests__/inventory/order/order.js [hunk 1]
```diff
@@ -143,6 +152,149 @@ describe("/store/carts", () => {
 
     describe("Fulfillments", () => {
       const lineItemId = "line-item-id"
+
+      it.only("Adjusts reservations on successful fulfillment on updated reservation item", async () => {
```
Comment (riqwan) [inline integration-tests/plugins/__tests__/inventory/order/order.js:156]: missed an `only` spot here.

**Review by riqwan — APPROVED** [review]
apart from the test, lgtm

**Review by adrien2p — APPROVED** [review]
LGTM

**Pull request merged**